### DISCUSSION TITLE
How do I write a function that returns something that has the signature of its own signature

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/csharplang#6397] [Language Ideas] [answered]
Such function, for example is named ``` Job ```
Would return something say ``` Rval ``` that can be called like ``` Job ``` can be called
Therefore we can see things like ``` var Rval = Job (MyParameter); Rval (MyParameter) (MyParameter) ... ```
To put it in another way we can for example
see :
```
Console.WriteLine("Hello") ("World") (GetPerson().ID)
```

**Comment by qrli** [comment, marked as answer]
```cs
Job<string> Foo(string parameter)
{
    Console.WriteLine(parameter);
    return Foo;
}
delegate Job<T> Job<T>(T parameter);

// Usage:
Foo("Hello")("World")("Whatever");
```

**Reply by Xyncgas** [reply]
You are amazing
